Building the documentation may fail

Follows: #1830 

**Component**:

'build', 'ci', 'documentation'

**What happened**:

When building the ISO, and specifically the documentation part, we can hit this error:

```
.  BUILD RPM REPO _build/root/packages/redhat/metalk8s-saltstack-el7/repodata/repomd.xml
.  BUILD RPM REPO _build/root/packages/redhat/metalk8s-extras-el7/repodata/repomd.xml
.  BUILD RPM REPO _build/root/packages/redhat/metalk8s-updates-el7/repodata/repomd.xml
.  BUILD RPM REPO _build/root/packages/redhat/metalk8s-epel-el7/repodata/repomd.xml
.  BUILD SRPM     _build/packages/redhat/metalk8s-sosreport-2.4-1.el7.src.rpm
TaskError - taskid:doc:pdf
Command '['/entrypoint.sh', 'latexpdf']' in image 'metalk8s-doc-builder:latest' returned non-zero exit status 1: b'':
b''
program finished with exit code 2
elapsedTime=793.921439
```

See this build: https://eve.devsca.com/github/scality/metalk8s/#/builders/1/builds/12658

**What was expected**:

The documentation should have been built smoothly.

**Steps to reproduce**

Not sure since it's a flaky. But it seems it happens a lot on 2.4 integration branches.

**Resolution proposal** (optional):

Not sure.

Now that we have proper output, here is the error I got (see https://eve.devsca.com/github/scality/metalk8s/#/builders/1/builds/13016):

```
docs run-test: commands[1] | /usr/src/metalk8s/docs/build.cmd latexpdf
make: Entering directory '/usr/src/metalk8s/docs'
Running Sphinx v2.0.1
Initializing Spelling Checker
Initializing GitHub plugin         
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [latex]: all documents
updating environment: 45 added, 0 changed, 0 removed
reading sources... [ 20%] developer/architecture/deployment .. developer/building/index
reading sources... [ 40%] developer/building/requirements .. developer/solutions/deploy
reading sources... [ 60%] developer/solutions/index .. operation/bootstrap_backup_restore
reading sources... [ 80%] operation/changing_node_hostname .. quickstart/bootstrap
reading sources... [100%] quickstart/expansion .. quickstart/setup
                                                                    
waiting for workers...
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
processing MetalK8s.tex... index-latex quickstart/index quickstart/introduction quickstart/setup quickstart/bootstrap quickstart/expansion quickstart/services installation-guide/index installation-guide/sizing operation/index operation/bootstrap_backup_restore operation/pr
eparation operation/upgrade operation/downgrade operation/changing_node_hostname operation/volume_management/index operation/volume_management/storageclass_creation operation/volume_management/volume_creation_deletion_cli operation/volume_management/volume_creation_deletio
n_gui developer/index developer/architecture/index developer/architecture/deployment developer/architecture/monitoring developer/architecture/requirements developer/design/index developer/design/volume_v1.0 developer/building/index developer/building/requirements developer
/building/building developer/building/configuration developer/building/features developer/running/index developer/running/cluster developer/running/storage_operator developer/running/ui developer/best_practices/index developer/best_practices/python developer/solutions/inde
x developer/solutions/introduction developer/solutions/archive developer/solutions/operator developer/solutions/deploy glossary 
resolving references...
done                                                                                                                                                                                                                                                                             
writing... failed

Exception occurred:
  File "/tmp/tox/docs/lib/python3.6/site-packages/sphinxcontrib/plantuml.py", line 421, in latex_visit_plantuml
    self.builder.warn(str(err))
AttributeError: 'LaTeXBuilder' object has no attribute 'warn'
The full traceback has been saved in /tmp/sphinx-err-a1db40z1.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:59: recipe for target 'latexpdf' failed
make: *** [latexpdf] Error 1
make: Leaving directory '/usr/src/metalk8s/docs'
ERROR: InvocationError for command /usr/src/metalk8s/docs/build.cmd latexpdf (exited with code 2)
___________________________________ summary ____________________________________
ERROR:   docs: commands failed
```
Looks related to `plantuml`…

```
docs installed: alabaster==0.7.12,argh==0.26.2,Babel==2.7.0,certifi==2019.9.11,chardet==3.0.4,colorama==0.4.1,doc8==0.8.0,docutils==0.15.2,httplib2==0.12.3,idna==2.8,imagesize==1.1.0,Jinja2==2.10.3,livereload==2.6.1,MarkupSafe==1.1.1,packaging==19.2,pathtools==0.1.2,pbr==5.4.3,plantuml==0.2.1,port-for==0.3.1,pyenchant==2.0.0,Pygments==2.4.2,pyparsing==2.4.2,pytz==2019.3,PyYAML==5.1.2,requests==2.22.0,restructuredtext-lint==1.3.0,six==1.12.0,snowballstemmer==2.0.0,Sphinx==2.2.0,sphinx-autobuild==0.7.1,sphinx-rtd-theme==0.4.3,sphinxcontrib-applehelp==1.0.1,sphinxcontrib-devhelp==1.0.1,sphinxcontrib-github-alt==1.1,sphinxcontrib-htmlhelp==1.0.2,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-plantuml==0.17.1,sphinxcontrib-qthelp==1.0.2,sphinxcontrib-serializinghtml==1.1.3,sphinxcontrib-spelling==4.3.0,stevedore==1.31.0,tornado==6.0.3,urllib3==1.25.6,watchdog==0.9.0
docs run-test-pre: PYTHONHASHSEED='1958304219'
docs run-test: commands[0] | doc8 docs/
Scanning...
Validating...
========
Total files scanned = 46
Total files ignored = 1
Total accumulated errors = 0
Detailed error counts:
    - CheckCarriageReturn = 0
    - CheckIndentationNoTab = 0
    - CheckMaxLineLength = 0
    - CheckNewlineEndOfFile = 0
    - CheckTrailingWhitespace = 0
    - CheckValidity = 0
docs run-test: commands[1] | /usr/src/metalk8s/docs/build.cmd latexpdf
make: Entering directory '/usr/src/metalk8s/docs'
Running Sphinx v2.2.0
Initializing Spelling Checker
Initializing GitHub plugin
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [latex]: all documents
updating environment: [new config] 45 added, 0 changed, 0 removed
reading sources... [ 20%] developer/architecture/deployment .. developer/building/index
reading sources... [ 40%] developer/building/requirements .. developer/solutions/deploy
reading sources... [ 60%] developer/solutions/index .. operation/bootstrap_backup_restore
reading sources... [ 80%] operation/changing_node_hostname .. quickstart/bootstrap
reading sources... [100%] quickstart/expansion .. quickstart/setup

waiting for workers...
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
processing MetalK8s.tex... index-latex quickstart/index quickstart/introduction quickstart/setup quickstart/bootstrap quickstart/expansion quickstart/services installation-guide/index installation-guide/sizing operation/index operation/bootstrap_backup_restore operation/preparation operation/upgrade operation/downgrade operation/changing_node_hostname operation/volume_management/index operation/volume_management/storageclass_creation operation/volume_management/volume_creation_deletion_cli operation/volume_management/volume_creation_deletion_gui developer/index developer/architecture/index developer/architecture/deployment developer/architecture/monitoring developer/architecture/requirements developer/design/index developer/design/volume_v1.0 developer/building/index developer/building/requirements developer/building/building developer/building/configuration developer/building/features developer/running/index developer/running/cluster developer/running/storage_operator developer/running/ui developer/best_practices/index developer/best_practices/python developer/solutions/index developer/solutions/introduction developer/solutions/archive developer/solutions/operator developer/solutions/deploy glossary 
resolving references...
done
writing... failed

Warning, treated as error:
error while running plantuml

b'Killed\n'
Makefile:59: recipe for target 'latexpdf' failed
make: *** [latexpdf] Error 1
make: Leaving directory '/usr/src/metalk8s/docs'
ERROR: InvocationError for command /usr/src/metalk8s/docs/build.cmd latexpdf (exited with code 2)
___________________________________ summary ____________________________________
ERROR:   docs: commands failed
```
https://eve.devsca.com/github/scality/metalk8s/#/builders/10/builds/7187

Hum, seems like my hunch was correct then.

I think we have resources issue here (that would explain the flakiness): we probably run out of memory.

I get the same error when running the container with low memory:

```
% docker run --rm --memory="128m"                     \
  --env TARGET_UID=$(id -u) --env TARGET_GID=$(id -g) \
  -v $(pwd)/docs/entrypoint.sh:/entrypoint.sh:ro      \
  -v $(pwd):/usr/src/metalk8s                         \
  metalk8s-doc-builder:latest /entrypoint.sh latexpdf
[…]
resolving references...
done
writing... failed

Warning, treated as error:
error while running plantuml

b'Killed\n'
Makefile:59: recipe for target 'latexpdf' failed
make: *** [latexpdf] Error 1
make: Leaving directory '/usr/src/metalk8s/docs'
ERROR: InvocationError for command /usr/src/metalk8s/docs/build.cmd latexpdf (exited with code 2)
___________________________________ summary ____________________________________
ERROR:   docs: commands failed
```

Maybe there is a way to reserve a minimal amount of memory upfront for the container (I guess this would still fail though, but maybe with a less cryptic error message).

Edit: seems not possible…
So maybe we can somehow raise the memory available during the build step (dunno how to do that, we aren't using a VM here…)

--- a/eve/workers/pod-builder/pod.yaml
+++ b/eve/workers/pod-builder/pod.yaml
@@ -17,10 +17,10 @@ spec:
           # There's a limit of 4 CPUs, so we assign one to `doit`, and 3 to
           # Docker.
           cpu: "1"
-          memory: 1Gi
+          memory: 2Gi
         limits:
           cpu: "1"
-          memory: 1Gi
+          memory: 2Gi
       env:
         - name: DOCKER_HOST
           value: localhost:2375
@@ -32,10 +32,10 @@ spec:
       resources:
         requests:
           cpu: "3"
-          memory: 2Gi
+          memory: 3Gi
         limits:
           cpu: "3"
-          memory: 2Gi
+          memory: 3Gi
       env:
         - name: DOCKER_TLS_CERTDIR
           value: ''

